src/tests/quantuminspire/qiskit/test_backend_qx.py

```diff
@@ -171,8 +178,44 @@ def test_get_experiment_results_returns_correct_value(self):
         job = QIJob('backend', '42', api)
         simulator = QuantumInspireBackend(api, Mock())
         experiment_result = simulator.get_experiment_results(job)[0]
-        self.assertEqual(experiment_result.data.to_dict()['counts']['0x1'], 50)
+        self.assertEqual(experiment_result.data.to_dict()['counts']['0x1'], 60)
         self.assertEqual(experiment_result.data.to_dict()['counts']['0x3'], 40)
+        self.assertEqual(len(experiment_result.data.to_dict()['memory']), 100)
+        self.assertEqual(experiment_result.data.to_dict()['memory'].count('0x1'), 60)
+        self.assertEqual(experiment_result.data.to_dict()['memory'].count('0x3'), 40)
+        self.assertEqual(experiment_result.name, 'circuit0')
+        self.assertEqual(experiment_result.shots, number_of_shots)
+
+    @staticmethod
+    def __api_get_return_values_single_shot(url):
+        if url == 'http://saevar-qutech-nginx/api/jobs/24/result/':
+            return {'id': 1, 'histogram': {'0': 0.5, '3': 0.5}, 'execution_time_in_seconds': 2.1,
+                    'number_of_qubits': 2, 'raw_data_url': 'http://saevar-qutech-nginx/api/results/24/raw-data/'}
+        return None
+
+    def test_get_experiment_results_returns_single_shot(self):
+        number_of_shots = 1
+        self._basic_job_dictionary['number_of_shots'] = number_of_shots
+        instructions = [{'name': 'h', 'params': [], 'texparams': [], 'qubits': [0]},
+                        {'name': 'cx', 'params': [], 'texparams': [], 'qubits': [0, 1]},
+                        {'name': 'measure', 'qubits': [1], 'memory': [1]},
+                        {'name': 'measure', 'qubits': [0], 'memory': [0]}]
+        experiment = self._instructions_to_two_qubit_experiment(instructions)
+        api = Mock()
+        api.get.side_effect = self.__api_get_return_values_single_shot
```

Comment:
with return_value set, each get on the api gets the same return value, with side_effect the function is called and there the url can be checked to see if the right url was passed, or depending on the url another return value can be returned when get was called more than once within the test. For the last reason I introduced this because get was called for getting the result and for getting the raw_data. Since the latest update, the get methode is now called once because we made an get_raw_data methode on the api to get the raw_data, so setting return_value is enough.